config.xsd

```diff
@@ -21,42 +21,43 @@
-        <xs:attribute name="memoizeMethodCallResults" type="xs:string" />
-        <xs:attribute name="hoistConstants" type="xs:string" />
-        <xs:attribute name="addParamDefaultToDocblockType" type="xs:string" />
-        <xs:attribute name="checkForThrowsDocblock" type="xs:string" />
-        <xs:attribute name="checkForThrowsInGlobalScope" type="xs:string" />
-        <xs:attribute name="forbidEcho" type="xs:string" />
         <xs:attribute name="errorBaseline" type="xs:string" />
-        <xs:attribute name="findUnusedCode" type="xs:string" />
-        <xs:attribute name="findUnusedVariablesAndParams" type="xs:string" />
-        <xs:attribute name="parseSql" type="xs:string" />
         <xs:attribute name="maxStringLength" type="xs:string" />
-        <xs:attribute name="resolveFromConfigFile" type="xs:string" />
-        <xs:attribute name="includePhpVersionsInErrorBaseline" type="xs:string" />
-        <xs:attribute name="loadXdebugStub" type="xs:string" />
-        <xs:attribute name="ensureArrayStringOffsetsExist" type="xs:string" />
-        <xs:attribute name="ensureArrayIntOffsetsExist" type="xs:string" />
+        <xs:attribute name="name" type="xs:string" />
+        <xs:attribute name="serializer" type="xs:string" />
+
+        <xs:attribute name="addParamDefaultToDocblockType" type="xs:boolean" default="false" />
+        <xs:attribute name="allowCoercionFromStringToClassConst" type="xs:boolean" default="false" />
+        <xs:attribute name="allowFileIncludes" type="xs:boolean" default="true" />
+        <xs:attribute name="allowPhpStormGenerics" type="xs:boolean" default="false" />
+        <xs:attribute name="allowStringToStandInForClass" type="xs:boolean" default="false" />
+        <xs:attribute name="checkForThrowsDocblock" type="xs:boolean" default="false" />
+        <xs:attribute name="checkForThrowsInGlobalScope" type="xs:boolean" default="false" />
+        <xs:attribute name="ensureArrayIntOffsetsExist" type="xs:boolean" default="false" />
+        <xs:attribute name="ensureArrayStringOffsetsExist" type="xs:boolean" default="false" />
+        <xs:attribute name="findUnusedCode" type="xs:boolean" default="false" />
+        <xs:attribute name="findUnusedVariablesAndParams" type="xs:boolean" default="false" />
+        <xs:attribute name="forbidEcho" type="xs:boolean" default="false" />
+        <xs:attribute name="hideExternalErrors" type="xs:boolean" default="false" />
+        <xs:attribute name="hoistConstants" type="xs:boolean" default="false" />
+        <xs:attribute name="ignoreInternalFunctionFalseReturn" type="xs:boolean" default="false" />
+        <xs:attribute name="ignoreInternalFunctionNullReturn" type="xs:boolean" default="true" />
+        <xs:attribute name="includePhpVersionsInErrorBaseline" type="xs:boolean" default="false" />
+        <xs:attribute name="loadXdebugStub" type="xs:boolean" /><!-- default is runtime-specific -->
```

Comment:
My xsd knowledge is a bit rusty, but i think you can add some kind of 'notes' to these things. 
Generally IDEs will then dislay those on hover or something.

I think this note about it being dependant on run time could be nice there.

Reply:
Not sure what you mean, quickly checked and there's nothing like that AFAIK.

Maybe any searchable hints? :)

Reply:
This?
https://www.w3schools.com/xml/el_annotation.asp

Reply:
Right, it does work, nice!

Reply:
Yeah it looks like that was the one i was thinking about, but it looks like that is only for top level. So i guess i miss remembered

Reply:
No, it actually does work:
![Screenshot from 2019-10-25 12-26-01](https://user-images.githubusercontent.com/209225/67564162-a58de780-f722-11e9-9f2b-0a639b852406.png)

Reply:
Thats pretty great 😄

Reply:
Thanks for the review! <3